### PR TITLE
Fix handling of EOF in insert_semicolon

### PR DESCRIPTION
**Pull request by ruben-ayrapetyan**

### CODE REVIEW

File: jerry-core/parser/js/parser.cpp [hunk 1]
```diff
@@ -2504,15 +2504,12 @@ insert_semicolon (void)
 {
   // We cannot use tok, since we may use lexer_save_token
   skip_token ();
-  if (token_is (TOK_NEWLINE) || lexer_prev_token ().type == TOK_NEWLINE)
+  if (token_is (TOK_NEWLINE)
```
Comment (egavrin) [inline jerry-core/parser/js/parser.cpp]:
```cpp
if (token_is (TOK_NEWLINE) || token_is (TOK_CLOSE_BRACE) || token_is (TOK_EOF)
|| (lexer_prev_token ().type == TOK_NEWLINE) )
```

Comment (ruben-ayrapetyan) [inline jerry-core/parser/js/parser.cpp]: I've updated the `if` block. Please, check.

**Comment by ruben-ayrapetyan** [comment]
@egavrin, pull request is updated. Please, check.

**Comment by egavrin** [comment]
Good to me

**Comment by zherczeg** [comment]
LGTM
make push

**Pull request merged**